src/main/java/eu/dissco/core/translator/terms/DigitalObjectDirector.java

```diff
@@ -0,0 +1,595 @@
+        .withDwcMaximumDepthInMeters(parseToDouble(new MaximumDepthInMeters(), data, dwc))
+        .withDwcMaximumDistanceAboveSurfaceInMeters(
+            parseToDouble(new MaximumDistanceAboveSurfaceInMeters(), data, dwc))
+        .withDwcMaximumElevationInMeters(parseToDouble(new MaximumElevationInMeters(), data, dwc))
+        .withDwcMinimumDepthInMeters(parseToDouble(new MinimumDepthInMeters(), data, dwc))
+        .withDwcMinimumDistanceAboveSurfaceInMeters(
+            parseToDouble(new MinimumDistanceAboveSurfaceInMeters(), data, dwc))
+        .withDwcMinimumElevationInMeters(parseToDouble(new MinimumElevationInMeters(), data, dwc))
+        .withDwcVerticalDatum(termMapper.retrieveTerm(new VerticalDatum(), data, dwc))
+        .withDwcLocationAccordingTo(termMapper.retrieveTerm(new LocationAccordingTo(), data, dwc))
+        .withDwcLocationRemarks(termMapper.retrieveTerm(new LocationRemarks(), data, dwc))
+        .withGeoreference(georeference)
+        .withGeologicalContext(geologicalContext);
+    var assertions = new OccurrenceAssertions().gatherOccurrenceAssertions(mapper, data, dwc);
+    var occurrence = new eu.dissco.core.translator.schema.Occurrences()
+        .withDwcFieldNumber(termMapper.retrieveTerm(new FieldNumber(), data, dwc))
+        .withDwcEventDate(termMapper.retrieveTerm(new DateCollected(), data, dwc))
+        .withDwcSex(termMapper.retrieveTerm(new Sex(), data, dwc))
+        .withDwcOrganismQuantity(termMapper.retrieveTerm(new OrganismQuantity(), data, dwc))
+        .withDwcOrganismQuantityType(termMapper.retrieveTerm(new OrganismQuantityType(), data, dwc))
+        .withDwcSamplingProtocol(termMapper.retrieveTerm(new SamplingProtocol(), data, dwc))
+        .withDwcLifeStage(termMapper.retrieveTerm(new LifeStage(), data, dwc))
+        .withDwcEventRemarks(termMapper.retrieveTerm(new EventRemark(), data, dwc))
+        .withDwcFieldNumber(termMapper.retrieveTerm(new FieldNumber(), data, dwc))
+        .withDwcFieldNotes(termMapper.retrieveTerm(new FieldNotes(), data, dwc))
+        .withDwcHabitat(termMapper.retrieveTerm(new Habitat(), data, dwc))
+        .withDwcReproductiveCondition(
+            termMapper.retrieveTerm(new ReproductiveCondition(), data, dwc))
+        .withDwcBehavior(termMapper.retrieveTerm(new Behavior(), data, dwc))
+        .withDwcEstablishmentMeans(termMapper.retrieveTerm(new EstablishmentMeans(), data, dwc))
+        .withDwcPathway(termMapper.retrieveTerm(new Pathway(), data, dwc))
+        .withDwcDegreeOfEstablishment(
+            termMapper.retrieveTerm(new DegreeOfEstablishment(), data, dwc))
+        .withDwcDegreeOfEstablishment(
+            termMapper.retrieveTerm(new GeoreferenceVerificationStatus(), data, dwc))
+        .withDwcOccurrenceStatus(termMapper.retrieveTerm(new OccurrenceStatus(), data, dwc) != null
+            ? DwcOccurrenceStatus.fromValue(
+            termMapper.retrieveTerm(new OccurrenceStatus(), data, dwc).toLowerCase()) : null)
```

Comment:
No it is tricky, have to think of something smart. The retrieveTerm() function is a generic function which returns a String. It first checks the mapping object and otherwise uses overridden methods which all return Strings. However, I like to keep the enums in the json-schema so I got a bit stuck here. For numeric value I have this method to parse the String value to an numeric value. However, for the enumerations this is more difficult, tried to make it work with generics but it was difficult. Can have another go, or see if github-copilot has ideas

Reply:
oh nifty